src/common.scss

```diff
@@ -0,0 +1,42 @@
+@import './variables.scss';
+
+.button {
+    padding: 5px;
+    border: unset;
+    border-radius: 5px;
+    text-transform: uppercase;
+    font-weight: bold;
+    cursor: pointer;
+}
+
+.button--magenta {
+    background-color: $magentaPink;
+    color: white;
+}
+
+.button--grey {
+    background-color: $grey1;
+}
+
+.button--selector {
+    background-color: transparent;
+    font-weight: normal;
+    color: white;
+    border-radius: 0;
+}
+
+.input {
+    padding: 10px;
+    border: unset;
+    border-radius: 5px;
+    color: white;
+}
+
+.input--darkGrey {
```

Comment:
In the name of BEM, and general class-naming conventions, it might be better to keep it consistent: `.input--dark-grey`. I assume currently used this way, it might be easier for you to interpolate classnames with `classnames`, like:
```
<div className={cn(`.input--${props.accent}`)}>
```

If using CSS modules, however, I would recommend to use `classnames` and imported styles like so, to have the better of **both BEM naming and CSS modules**:

```
import cn from 'classnames/bind';
import style from './style.module.scss';
const cx = cn.bind(style);

enum Accents {
  DARK_GREY: 'dark-grey',
  BURGUNDY_RED: 'burgundy-red'
}

<div className={cx({ 
  input: !props.accent,                      // If we DO NOT receive an Accents enum in props.accent, just put ".input"
  [`input--${props.accent}`]: props.accent   // Else interpolate. Given props.accent = Accents.DARK_GREY => ".input--dark-grey"
})}>
```

The approach above showcases [this](https://www.npmjs.com/package/classnames#alternate-bind-version-for-css-modules-) feature of `classnames`

Reply:
A bit more extrapolated example here:

```typescript

enum Accent {
  RED: 'red',
  BLUE: 'blue'
}

type ButtonProps = {
  accent?: Accent;
};

// OR if you want, you can save the enum declaration, by going simply:
type ButtonProps = {
  accent?: 'red' | 'blue'; // This could be also extracted to eg.: type ButtonColors = 'red' | 'blue';
};

const Button = ({ accent }: ButtonProps) => (
  <button className={cx(accent ? `btn--${accent}` : 'btn')}>Test</button>
)

const elementWthAccent = <Button accent={Accent.RED} />;
// Renders <button class="btn--red">Test</button>
// Also note the type-safety
const element = <Button />; // Renders <button class="btn">Test</button>
```